mmcls/models/backbones/swin_transformer.py

```diff
@@ -425,3 +430,26 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, *args,
 
         super()._load_from_state_dict(state_dict, prefix, local_metadata,
                                       *args, **kwargs)
+
+    def _freeze_stages(self):
+        if self.frozen:
+            self.patch_embed.eval()
+            for param in self.patch_embed.parameters():
+                param.requires_grad = False
+
+        for i in range(len(self.stages)):
+            m = self.stages[i]
+            m.eval()
+            for param in m.parameters():
+                param.requires_grad = False
+
+    def train(self, mode=True):
+        super(SwinTransformer, self).train(mode)
+        self._freeze_stages()
+        if mode and self.norm_eval:
+            for m in self.modules():
+                # trick: eval have effect on BatchNorm only
+                if isinstance(m, _BatchNorm):
+                    m.eval()
+                if isinstance(m, nn.modules.LayerNorm):
+                    m.eval()
```

Comment:
Does `LayerNorm` need to turn to evaluation mode? Unlike `BatchNorm`, `LayerNorm` should have the same behavior in both training mode and eval mode.

Reply:
It also puzzles me.  I write this reference https://github.com/open-mmlab/mmclassification/blob/9d9dce69ad1d7911c87c69e30bce4a928f4dcd94/mmcls/models/backbones/resnet.py#L632 .